modules/httpcontrol/index.js

```diff
@@ -0,0 +1,123 @@
+import { EventEmitter } from 'events';
+import http from 'http';
+import querystring from 'querystring';
+
+/**
+ * Receives http requests from another peer.
+ */
+class HttpControl extends EventEmitter {
+    /**
+     * Construcs new instance.
+     * 
+     * @class
+     */
+    constructor() {
+        super();
+        this.started = false;
+        this.server = null;
+    }
+
+    /**
+     * Initializes the http server.
+     * 
+     * @param {integer} port - listening port for http server
+     * @returns {void}
+     */
+    init(port) {
+        this.server = http.createServer((req, res) => {
+            if (this.started) {
+                /*  HTTP request format: 
+        curl --data "command=<command.type>&args=<arguments>" <targetURL>
+                 */
+                console.log('HTTP request received');
+                let body = '';
+                let query;
+
+                req.on('data', data => {
+                    body += data;
+                });
+                req.on('end', () => {
+                    query = querystring.parse(body);
+                    console.log('Received request: ');
+                    console.log(query);
+
+                    this.on('response', (success, message) => {
+                        if (success) {
+                            console.log('Command execution success');
+                            res.writeHead(200);
+                            res.end(message);
+                        } else {
+                            console.log('Command execution failed');
+                            res.writeHead(400);
+                            res.end(message);
+                        }
+                    });
+
+                    this.emit('command', query.command, query.args);
+                });
+            } else {
+                console.log('Server disabled');
+                res.writeHead(503);
+                res.end('Service Unavailable\n');
+            }
+        });
+        this.server.listen(port, () => {
+            console.log('Server listening at port %d', port);
+        });
+    }
+
+    /**
+     * Starts processing the events.
+     * 
+     * @returns {void}
+     */
+    start() {
```

Comment:
I would remove them and put the start logic into the constructor. You can also do a dispose method if there is something that need to be cleaned up. This will cause exporting the class and creating the object where you are calling start now.